src/cmdstan/command.hpp

```diff
@@ -139,154 +154,66 @@ int command(int argc, const char *argv[]) {
-          << " and output file names are identical, must be different."
-          << std::endl;
-      throw std::invalid_argument(msg.str());
-    }
-  }
-
-  // Open output streams
-  stan::callbacks::writer init_writer;
-  stan::callbacks::interrupt interrupt;
+  // General config for all methods
+  unsigned int id = get_arg_val<int_argument>(parser, "id");
+  int sig_figs = get_arg_val<int_argument>(parser, "output", "sig_figs");
+  int refresh = get_arg_val<int_argument>(parser, "output", "refresh");
 
   //////////////////////////////////////////////////
   //                Initialize Model              //
   //////////////////////////////////////////////////
+  arg_seed *random_arg
+      = dynamic_cast<arg_seed *>(parser.arg("random")->arg("seed"));
+  unsigned int random_seed = random_arg->random_value();
+
   std::string filename = get_arg_val<string_argument>(parser, "data", "file");
 
   std::shared_ptr<stan::io::var_context> var_context
       = get_var_context(filename);
 
-  // Instantiate model
   stan::model::model_base &model
       = new_model(*var_context, random_seed, &std::cout);
 
-  std::vector<std::string> model_compile_info = model.model_compile_info();
-
-  //////////////////////////////////////////////////
-  //                Initialize Writers            //
-  //////////////////////////////////////////////////
-
-  std::string output_file
-      = get_arg_val<string_argument>(parser, "output", "file");
-  if (output_file == "") {
-    throw std::invalid_argument(
-        std::string("File output name must not be blank"));
-  }
-  std::string output_name;
-  std::string output_ending;
-  size_t output_marker_pos = output_file.find_last_of(".");
-  if (output_marker_pos > output_file.size()) {
-    output_name = output_file;
-    output_ending = "";
-  } else {
-    output_name = output_file.substr(0, output_marker_pos);
-    output_ending = output_file.substr(output_marker_pos, output_file.size());
-  }
-
-  std::string diagnostic_file
-      = get_arg_val<string_argument>(parser, "output", "diagnostic_file");
-  size_t diagnostic_marker_pos = diagnostic_file.find_last_of(".");
-  std::string diagnostic_name;
-  std::string diagnostic_ending;
-  // no . seperator found.
-  if (diagnostic_marker_pos > diagnostic_file.size()) {
-    diagnostic_name = diagnostic_file;
-    diagnostic_ending = "";
-  } else {
-    diagnostic_name = diagnostic_file.substr(0, diagnostic_marker_pos);
-    diagnostic_ending
-        = diagnostic_file.substr(diagnostic_marker_pos, diagnostic_file.size());
-  }
-
-  std::vector<stan::callbacks::unique_stream_writer<std::ostream>>
-      sample_writers;
-  sample_writers.reserve(num_chains);
-  std::vector<stan::callbacks::unique_stream_writer<std::ostream>>
-      diagnostic_writers;
-  diagnostic_writers.reserve(num_chains);
+  // Setup callbacks
+  stan::callbacks::interrupt interrupt;
+  stan::callbacks::writer dummy_writer;
```

Comment:
Because it's currently unusued it does not matter, but I suspect if it ever becomes used the Multi-path pathfinder service will expect a structured writer here rather than a `writer`

Reply:
changed to json_writer.